src/main/java/org/cryptomator/frontend/fuse/ReadOnlyFileHandler.java

```diff
@@ -92,7 +93,12 @@ public int release(Path path, FuseFileInfo fi) {
 	}
 
 	public int getattr(Path node, BasicFileAttributes attrs, FileStat stat) {
-		stat.st_mode.set(FileStat.S_IFREG | 0444);
+		if (attrs instanceof PosixFileAttributes) {
+			long mode = attrUtil.posixPermissionsToOctalMode(((PosixFileAttributes) attrs).permissions());
```

Comment:
I prefer to have the cast in its own line. More legible. :wink:

```suggestion
			PosixFileAttributes posixAttrs = (PosixFileAttributes) attrs;
			long mode = attrUtil.posixPermissionsToOctalMode(posixAttrs.permissions());
```